File: TaskManager/Domain/Entities/Tasks.cs
﻿using System.ComponentModel.DataAnnotations;

namespace TaskManager.Domain.Entities
{
	public class Tasks 
	{
		[Key]
		public Guid TaskId { get; set; }

		public string Name { get; set; }
		public string Description { get; set; }
		public string? AssignedEmail { get; set; } = string.Empty;
		public string Priority { get; set; } = string.Empty;

		public Project Project { get; set; } = null!;

		public Tasks() { }

		public Tasks(Guid taskId, string name, string description, string assignedEmail, string priority, Project project)
		{
			TaskId = taskId;
			Name = name;
			Description = description;
			AssignedEmail = assignedEmail;
			Priority = priority;
			Project = project;
		}

	}
	
}

File: TaskManager/ClientApp/src/app/api/models/tasks.ts
/* tslint:disable */
/* eslint-disable */
import { Project } from './project';
export interface Tasks {
  assignedEmail?: null | string;
  description?: null | string;
  isComplete?: number;
  name?: null | string;
  priority?: null | string;
  project?: Project;
  taskId?: string;
}
